### DISCUSSION TITLE
LIOJ1028 生命靈數

### DISCUSSION THREAD

**Discussion by chadomdase** [Lidemy/forum#306] [程式導師實驗計畫]
```
import sys
import math

line = sys.stdin.readline()    
line = line.replace(" ", "")
line = [int(x) for x in line]
n = len(line)

def addDigits(n):
    sum = 0
    while n != 0:
        sum += n % 10
        n = math.floor(n / 10)
    return sum

def solve(line):
    num = 0
    for i in range(n):
        num += line[i]
    num = addDigits(num)
    while num >= 10:
        num = addDigits(num)
    print(num)

solve(line)
```

大家好，這段程式碼是用Python寫得，想知道這樣寫還是一直得到wrong answer的問題在哪QQ

**Comment by cwc329** [comment]
建議把 code 用 [markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code) 打出來，
用截圖在一些裝置上不好閱讀，
也會讓其他人很難實際跑你的 code 去除錯。

**Reply by cwc329** [reply]
可能是這個問題
[python argument shadowing](https://stackoverflow.com/questions/63880231/why-and-how-to-resolve-shadows-argument-name-from-outer-scope-in-class-python)
把變數命名的更好一些，避免 a,b,n,x 等名稱，也不要有 shadowing。

**Reply by chadomdase** [reply]
不好意思，已經改成markdown的方式。
但是我在vscode上執行結果答案都正確，上傳LIOJ就顯示WA，還有可能是argument shadowing的問題嗎?

**Reply by cwc329** [reply]
我對 python 的 runtime 不熟，
也有可能是你讀檔案的方式或者處理字串的地方出錯。
我自己是習慣用 cli 跑 `cat input.txt | node index.js` 處理，
你也可以用類似的方式跑看看，
我用這個方式使用範例測資跑你這份 code 答案也是錯的。

**Reply by chadomdase** [reply]
我也是在想會不會是讀檔的問題，還是謝謝您
